jetty-server/src/main/java/org/eclipse/jetty/server/HttpOutput.java

```diff
@@ -591,17 +591,18 @@ public void write(byte[] b, int off, int len) throws IOException
         // handle blocking write
 
         // Should we aggregate?
-        int capacity = getBufferSize();
+        // Yes - if the write is smaller than the commitSize (==aggregate buffer size)
+        //       and the write is not the last one, or is last but will fit in an already allocated aggregate buffer.
         boolean last = isLastContentToWrite(len);
-        if (!last && len <= _commitSize)
+        if (len <= _commitSize && (!last || len <= BufferUtil.space(_aggregate)))
         {
             acquireBuffer();
 
             // YES - fill the aggregate with content from the buffer
             int filled = BufferUtil.fill(_aggregate, b, off, len);
 
-            // return if we are not complete, not full and filled all the content
-            if (filled == len && !BufferUtil.isFull(_aggregate))
+            // return if we are not the last write and have aggregated all of the content
+            if (!last && filled == len && !BufferUtil.isFull(_aggregate))
```

Comment:
This `!last` was pointed out in the prior review, how is it still needed?

Reply:
As discussed on the hangout, this is needed because we can entere the loop with `!last`.  I restructured the if statement above to make that clear by having the `||` statement.